cmake/mcuboot.cmake

```diff
@@ -97,6 +103,7 @@ function(zephyr_mcuboot_tasks)
   if(CONFIG_BUILD_OUTPUT_BIN)
     list(APPEND unconfirmed_args --bin --sbin ${output}.signed.bin)
     list(APPEND byproducts ${output}.signed.bin)
+    zephyr_runner_file(bin ${output}.signed.bin)
```

Comment:
Hi @mbolivar , do you think have the version on output file name could bring some value? I mean, to make visible the value that we can expect to have on MCUboot headers.

Reply:
@nandojve sure, you can do that with something like this:

`west build ... -- -DCONFIG_KERNEL_BIN_NAME=my-firmware-1.0.bin -DCONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS=\"--version 1.0\"`

I know it's a bit of extra repetition, but I think it's the cleanest way to do it. Otherwise we have to parse the imgtool command line arguments from here, which would be too brittle for my taste.

Reply:
Can we access CMake variables from Kconfig? If so, the application CMakeLists.txt file could set the version in a variable and prj.conf use that variable for those two Kconfig variables.

Reply:
There's a restricted subset of CMake variables that are mirrored into Kconfig files, e.g. `ZEPHYR_<MODULE>_MODULE_DIR` and any other environment variables in `COMMON_KCONFIG_ENV_SETTINGS`. But there's nothing like that available in .conf fragments as far as I know; they contain literal values only.

Reply:
To be clear, the distinction in my previous comment is about files like prj.conf (literal values only) vs files like Kconfig.zephyr (some CMake variables mirrored into the environment).

Reply:
Adding something which does not provide name and version support will make it confusing for users where we'll have diverging ways of doing this. It'd be interesting to cover those use cases so we can have a single way of using this which covers most of the use cases.

Reply:
> Adding something which does not provide name and version support

As I showed above, there are already mechanisms for naming the file and setting its version.

> will make it confusing for users where we'll have diverging ways of doing this

I'm not sure I see a problem with divergent uses for the same mechanism. Am I missing something?

>  It'd be interesting to cover those use cases so we can have a single way of using this which covers most of the use cases.

I think the use cases you mention are already covered; can you explain what is missing and why it is in scope for this PR?

The point of this PR is just to make it possible for 'west build' and 'west flash' to work together properly when the application is to be chain-loaded by MCUboot. It doesn't change any default behavior from how things work in Zephyr 2.3. Users must explicitly use the new Kconfig options added in #27688 to opt-in to this. The documentation for these Kconfig options and the workflow in the .rst are up to date.

Reply:
Oh ok, I missed it. @mbolivar sorry for the noise.

Reply:
NP, please do let me know if something in scope is missing!